# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/nsNSSCallbacks.h
# Commit: fa57574775b2
# Full Hash: fa57574775b29f4a03667d9c350a1be269adfaf7
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2019-09-26 21:35:42
# Regressor Bug: 1577643
# File Overlap Count: 1
# Description:
#   Bug 1577643 - Implement a security info class for the quic transport. r=keeler
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44073
# ==============================================================================

diff -r 02f72f7040e4 -r fa57574775b2 security/manager/ssl/nsNSSCallbacks.h
--- a/security/manager/ssl/nsNSSCallbacks.h	Thu Sep 26 10:14:53 2019 +0000
+++ b/security/manager/ssl/nsNSSCallbacks.h	Wed Sep 25 20:23:56 2019 +0000
@@ -22,6 +22,7 @@
 using mozilla::Vector;
 
 class nsILoadGroup;
+class nsIX509CertList;
 
 char* PK11PasswordPrompt(PK11SlotInfo* slot, PRBool retry, void* arg);
 
@@ -35,4 +36,9 @@
     size_t ocspRequestLength, TimeDuration timeout,
     /*out*/ Vector<uint8_t>& result);
 
+nsCString getKeaGroupName(uint32_t aKeaGroup);
+nsCString getSignatureName(uint32_t aSignatureScheme);
+nsresult IsCertificateDistrustImminent(nsIX509CertList* aCertList,
+                                       /* out */ bool& isDistrusted);
+
 #endif  // nsNSSCallbacks_h